### PR TITLE
Reestructure project

### PR DESCRIPTION
**Pull request by adrianfusco**
- Updated the name of the project from `gz-beat-shipper` to `beatshipper` in GitHub and all the references in the project.
- Update README with the information based on the actual status of the project
- Now it can check multiple paths:
	- Modified `configs/config.go` the `Configuration struct` to allow a slice of paths
- Added the possibility to process no compressed files also, not just GNU Zip ones.
- Modified the `Registry` struct.
- Added minimal CLI functionality using the module `cobra`:
	- Added command `send`
- Moved the logic that was used to send the data to `pkg/shipper/shipper.go`. It had `GZ` references. Now we can read from no compressed files so the nomenclature of the functions are neutral without including any extension.

Summary:

- Add CLI functionality.
- Update README.md to actual status of the project
- Allow explode multiple paths
- Allow process no compressed files instead of just GNU Zip ones
- Changed structure of the JSON registry. Added `date` field instead of just `name` of the files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,89 @@
-# gz-beat-shipper
+# beatshipper
 
 Since there’s no way to send GNU zip files through filebeat, this service will be responsible for checking if there are new .gz files based on a path that we’ll explode using globbing, decompress them, send them using the filebeat service with the provided configuration and store them in a local file registry.
+
+It can be used also for not compressed files.
```
Comment (jsanemet) [inline README.md]: 'It can **also** be used for **non-compressed** files.'

File: README.md [hunk 2]
```diff
@@ -1,3 +1,89 @@
...
+
+[![License](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+## Requirements
+
+[Download](https://golang.org/dl/) and install **Go**.
```
Comment (jsanemet) [inline README.md]: Well, the compiler as a requirement is a given. By pointing to the language version should be enough.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+Flags:
+  -h, --help      help for beatshipper
+  -v, --version   version for beatshipper
+
+Use "beatshipper [command] --help" for more information about a command.
+```
+
+## Configuration
+
+We should create the configuration in the following file `/etc/beatshipper/beatshipper-conf.yml`. An example is located in the root of the project. We should add the following fields:
```
Comment (jsanemet) [inline README.md:46]: What if I do not have root rights?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,89 @@
...
+
+- `host`
+- `port`
+- `path` array of paths that will be exploded using `GLOB`
+- `registry`: the name of the file where we'll store the processed files.
+- `recheck` time of recheck (re-process)
+
```
Comment (jsanemet) [inline README.md:53]: Have these as comments on the config sample as well.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,89 @@
...
+Available Commands:
+  completion  Generate the autocompletion script for the specified shell
+  help        Help about any command
+  send        Send the message as a beat
+
+Flags:
+  -h, --help      help for beatshipper
+  -v, --version   version for beatshipper
+
+Use "beatshipper [command] --help" for more information about a command.
```
Comment (jsanemet) [inline README.md:38]: Indicate version this help was generated from. That way you may alter it in the future without having to update this as well.

File: README.md [hunk 6]
```diff
@@ -1,3 +1,89 @@
...
+
+- `host`
+- `port`
+- `path` array of paths that will be exploded using `GLOB`
+- `registry`: the name of the file where we'll store the processed files.
+- `recheck` time of recheck (re-process)
+
+## Usage:
+
+- Manually:
```
Comment (jsanemet) [inline README.md]: As opposed to?

File: README.md [hunk 7]
```diff
@@ -1,3 +1,89 @@
...
+## Registry
+
+Files will be stored in the following way:
+
+```
+$ jq < shipper_registry.json
+{
+  "ParsedFiles": [
+    {
+      "name": "/rhos-infra-dev-netapp/jenkins-logs/rcj/test/sub-test/sub-sub-test/var/log/extra/rpm-list.txt.gz",
```
Comment (jsanemet) [inline README.md]: Be careful not to display anything from downstream.

File: main.go [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+//    License for the specific language governing permissions and limitations
+//    under the License.
+package main
+
+import (
+	"beatshipper/cmd/shipper"
+)
+
+func main() {
+	shipper.Execute()
```
Comment (jsanemet) [inline main.go]: That function returns a possible error you are just ignoring. You probably would like to print the stack trace or something against one.

**Review by jsanemet — APPROVED** [review]
Looks ok, just added some comments to fine tune it.

File: README.md (hunk 1 above)
Comment (adrianfusco) [reply README.md]: Oh, thanks. Change done. I knew the B2.2 of the English school wasn’t good at all :p

File: README.md (hunk 2 above)
Comment (adrianfusco) [reply README.md]: Changed, thank you.

File: README.md (hunk 4 above)
Comment (adrianfusco) [reply README.md:53]: Changed, thank you.

File: README.md (hunk 5 above)
Comment (adrianfusco) [reply README.md:38]: Added. Thank you.

File: README.md (hunk 6 above)
Comment (adrianfusco) [reply README.md]: I forgot to delete it. Thanks, done.

File: README.md (hunk 7 above)
Comment (adrianfusco) [reply README.md]: You're right. I added just a path with no sense but I've changed. Thanks.

**Review by IgnacioPro — APPROVED** [review]
+1

File: main.go (hunk 8 above)
Comment (adrianfusco) [reply main.go]: You're right. I've added it. Thank you.

File: README.md (hunk 3 above)
Comment (adrianfusco) [reply README.md:46]: I'll work on it for the next PR. Thanks for the reviews, Jorge.

**Pull request merged**